salt/states/file.py

```diff
@@ -452,13 +452,16 @@ def _check_directory(name,
                      mode,
                      clean,
                      require,
-                     exclude_pat):
+                     exclude_pat,
+                     max_depth=None):
     '''
     Check what changes need to be made on a directory
     '''
     changes = {}
     if recurse or clean:
-        walk_l = list(os.walk(name))  # walk path only once and store the result
+        assert(max_depth is None or not clean)
```

Comment:
This was meant to catch programming and not user errors.
I could not find an example of "exiting" in the source code. Simply returning here is not an option: I would need to return None instead of an empty list to indicate the failure. This would mean to change all call-sides as well.
We can add a message to the assert, though. (Or remove it if its not adding any benefit in your eyes.)